openapi/api_auth.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Authorisation API
+ *
+ * The subset of the [Authentication API as defined by EBSI](https://api-pilot.ebsi.eu/docs/apis/authorisation/latest#/).
+ *
+ * API version: v4
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package openapi
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/fiware/VCVerifier/logging"
+	"github.com/gin-gonic/gin"
+)
+
+var ErrorMessageUnableToDecodeBody = ProblemDetails{Title: "Code not decode body.", Status: 400}
+
+func GetTokenM2M(c *gin.Context) {
+	var requestBody TokenRequestBody
+	if err := c.BindJSON(&requestBody); err != nil {
+		logging.Log().Error("Could not map request body", err)
+		c.AbortWithStatusJSON(400, ErrorMessageUnableToDecodeBody)
+		return
+	}
+	bytes, err := base64.RawURLEncoding.DecodeString(requestBody.VpToken)
+	if err != nil {
+		logging.Log().Infof("Was not able to decode the form string %s. Err: %v", requestBody.VpToken, err)
+		c.AbortWithStatusJSON(400, ErrorMessageUnableToDecodeToken)
+		return
+	}
+	var rawCredentials []map[string]interface{}
+
+	err = json.Unmarshal(bytes, &rawCredentials)
+	if err != nil {
+		logging.Log().Infof("Was not able to decode the credentials from the token %s. Err: %v", requestBody.VpToken, err)
+		c.AbortWithStatusJSON(400, ErrorMessageUnableToDecodeCredential)
+		return
+	}
+	clientId := c.GetHeader("client_id")
+
+	scopes := strings.Split(requestBody.Scope, ",")
+
+	// TODO What to add as subject and audience?
```

Comment:
Spec says:
> sub (subject): Subject of the JWT (the user)
> aud (audience): Recipient for which the JWT is intended
So both the client/serviceId?

Reply:
Audience should be the client that the token is requested for. 

For subject, do we know the user? Since we have a verifiable presentation, we might have multiple VCs in the token generation and none of them necessarily is a user. We should probably not use it.